### ISSUE TITLE
Save dialog not being displayed for new objects when mutations have been made

### ISSUE THREAD

**Issue by larkin** [nasa/openmct#1080]
Detected in VISTA in a recently deployed version:

1. create new object (say, historical table)
2. add one or more telemetry elements to the table
3. press save

Expected behavior: the "save in" dialog will show and prompt the user for a name and location to save the element in.
Actual behavior: the object saves and is orphaned.

Assigning to myself to isolate further.

**Comment by larkin** [comment]
Was re-reported by a VISTA user, I spent more time fooling around and I have worked out a reliable reproduction.

The same sorts of steps work with multiple types of objects.  It is triggered initially by a browse mode drag and drop compose and then affects all objects created and mutated after that until a navigation event occurs.

To reproduce it, you need a few objects which you can compose into other objects.  I created two layouts, "layout 1" and "layout 2".  It also works with fixed position, telemetry panels, tables, etc.

1. In browse mode, navigate to "layout 2".
2. drag and drop "layout 1" into "layout 2" to compose it.
3. Press save.
4. Create a new layout, and don't exit edit mode.  
5. Observe that the save button says "save as".
6. Drag and drop "layout 1" (or "layout 2", doesn't matter) into the layout.
7. Observe that the save button now says "save".
8. Click the "save" button.
9. observe that you were not asked for a location to persist the object and that the object cannot be found.

It doesn't matter what object you are creating after step 3, though there seems to be some differences in the bug based on the object type

* Layout: "save as" becomes "save" after adding one object to layout.
* Fixed Position Display: "save as" becomes "save" after adding one object to display.
* Telemetry Panel (viewed as plot): becomes "save" after adding two objects to panel.
* Historical Telemetry Table: becomes "save" after adding two objects to table.

After completing step 3, the bug will continue to occur until you navigate to a different object via the tree.  It can be re-triggered without refreshing the page.

So, short story: 

1. drag and drop compose to trigger edit mode
2. save
3. create new thing
4. compose objects in new thing
5. save new thing
6. enjoy orphans

@akhenry or @VWoeltjen can you reproduce this?  It's easy to encounter when doing a lot of editing in the application and has stung me more than once.

**Comment by VWoeltjen** [comment]
Yes, able to reproduce in Open MCT on master by following steps above.

**Comment by VWoeltjen** [comment]
Debugging, noticed that `appliesTo` for SaveAs gets called twice when dropping into a layout (between steps 6 and 7 above.) The first time, it returns true (as expected); the second time, it returns false because `domainObject.getModel().persisted !== undefined`

Why `persisted` has an assigned value at that point is unclear, needs further investigation.

**Comment by akhenry** [comment]
[Created a PR](#1201) for this. The ultimate cause was that the `EditAction` [adds a listener](https://github.com/nasa/openmct/blob/e0b69868519e11910a828d9fc339dd1a8572794e/platform/commonUI/edit/src/actions/EditAction.js#L83) to the `NavigationService` that cancels edit mode if the user navigates away. [Canceling edit is asynchronous](https://github.com/nasa/openmct/blob/e0b69868519e11910a828d9fc339dd1a8572794e/platform/commonUI/edit/src/services/TransactionService.js#L144) though, so when the `CreateAction` was invoked, a new transaction was created which was subsequently [cancelled by a deferred callback](https://github.com/nasa/openmct/blob/e0b69868519e11910a828d9fc339dd1a8572794e/platform/commonUI/edit/src/services/TransactionService.js#L145).

With the transaction now canceled, [a persist call](https://github.com/nasa/openmct/blob/e0b69868519e11910a828d9fc339dd1a8572794e/platform/commonUI/edit/src/actions/EditAndComposeAction.js#L47) in the `EditAndComposeAction` caused the layout to be persisted immediately.

The solution was to move the cancellation of edit mode into the `NavigateAction` itself so that it happens synchronously when the user navigates away from an object being edited.

**Comment by larkin** [comment]
Testathon 2016-10-21: verified fixed.
